packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js

```diff
@@ -0,0 +1,237 @@
+// @ts-check
+import { assert, details } from '@agoric/assert';
+import { importBundle } from '@agoric/import-bundle';
+import { Remotable, getInterfaceOf, makeMarshal } from '@agoric/marshal';
+// grumble... waitUntilQuiescent is exported and closes over ambient authority
+import { waitUntilQuiescent } from '../../waitUntilQuiescent';
+
+import { makeLiveSlots } from '../liveSlots';
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+// eslint-disable-next-line no-unused-vars
+function workerLog(first, ...args) {
+  // @ts-ignore
+  // eslint-disable-next-line
+  // print(`---worker: ${first}`, ...args);
+}
+
+workerLog(`supervisor started`);
+
+/**
+ * @typedef { [unknown, ...unknown[]] } Tagged
+ */
+const Item = {
+  /** @type { (item: Tagged) => ArrayBuffer } */
+  encode: tagged => encoder.encode(JSON.stringify(tagged)).buffer,
+
+  /** @type { (msg: ArrayBuffer) => Tagged } */
+  decode(msg) {
+    const txt = decoder.decode(msg);
+
+    /** @type { Tagged } */
+    const item = JSON.parse(txt);
+    assert(Array.isArray(item), details`expected array`);
+    assert(item.length > 0, details`empty array lacks tag`);
+    return item;
+  },
```

Comment:
`item` and `tagged` appear to be synonyms. Maybe `TaggedItemCodec` (or `codec`) and `taggedItem`  (or just `tag`) throughout?